### PR TITLE
Removed the data processor.

### PR DESCRIPTION
**Pull request by chathurapriyankara**

### CODE REVIEW

**Comment by chathurapriyankara** [comment]
This PR is fine. Please merge this.

**Comment by chathurapriyankara** [comment]
This PR is fine. Need to merge.

File: src/main/java/server/ServerThread.java [hunk 1]
```diff
@@ -2,42 +2,65 @@
...
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.PrintWriter;
 import java.net.Socket;
-import java.util.ArrayList;
 
 public class ServerThread extends Thread {
     private Socket socket;
-    private DataProcessor dataProcessor;
```
Comment (chathurauok) [inline src/main/java/server/ServerThread.java:16]: Why did you remove the data processor ?

File: src/main/java/server/ServerThread.java [hunk 2]
```diff
@@ -2,42 +2,65 @@
...
 
     public ServerThread(Socket s, TextArea textArea) {
         this.socket = s;
-        dataProcessor = new DataProcessor();
         this.textArea = textArea;
     }
 
     public void run() {
         //Try with resources
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
```
Comment (chathurauok) [inline src/main/java/server/ServerThread.java:27]: Why did you change the try with resources ?

Comment (DhanJaya) [reply src/main/java/server/ServerThread.java:27]: Following the best practices

Comment (chathurapriyankara) [reply src/main/java/server/ServerThread.java:27]: Reply

File: src/main/java/server/ServerThread.java (hunk 1 above)
Comment (DhanJaya) [reply src/main/java/server/ServerThread.java:16]: Since its not used

File: src/main/java/server/ServerThread.java (hunk 2 above)
Comment (chathurapriyankara) [reply src/main/java/server/ServerThread.java:27]: Reply2

File: src/main/java/server/ServerThread.java [hunk 3]
```diff
@@ -2,42 +2,65 @@
 
 import javafx.scene.control.TextArea;
 import parser.DataProcessor;
-import parser.Food;
```
Comment (chathurauok) [inline src/main/java/server/ServerThread.java:5]: Add this back.

**Review by chathurauok — CHANGES_REQUESTED** [review]

File: src/main/java/server/ServerThread.java [hunk 4]
```diff
@@ -2,42 +2,65 @@
...
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.PrintWriter;
 import java.net.Socket;
-import java.util.ArrayList;
```
Comment (chathurauok) [inline src/main/java/server/ServerThread.java:12]: You need the array list ?

**Review by chathurauok — COMMENTED** [review]
done

**Pull request merged**